hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/NameNodeRpcServer.java

```diff
@@ -2285,6 +2285,7 @@ public BatchedEntries<ZoneReencryptionStatus> listReencryptionStatus(
   public void setErasureCodingPolicy(String src, String ecPolicyName)
       throws IOException {
     checkNNStartup();
+    namesystem.checkOperation(OperationCategory.WRITE);
```

Comment:
Can you explain why `checkOperation` is added here? This has been judged in `namesystem#setErasureCodingPolicy`.

Reply:
@tomscut Sir, thanks for you review.

`checkOperation` here is used to judge the HA status at the first time. If the HA status does not meet the requirement,    an exception can be returned as soon as possible. 

You can see this usage in many methods.

Reply:
> @tomscut Sir, thanks for you review.
> 
> `checkOperation` here is used to judge the HA status at the first time. If the HA status does not meet the requirement, an exception can be returned as soon as possible.
> 
> You can see this usage in many methods.

I see. But it seems unrelated to `addBlock`? Maybe we should open a new jira to add this check to all missing methods?

Reply:
Yeah, not related to addBlock. They are similar problems, so I fixed them together in this PR.  I will create a new jira to split them later. Thanks so much.